### PR TITLE
Create GPUAddon CR on boot if not exists

### PR DESCRIPTION
**Pull request by sagidayan**
This PR Adds a jumpstart function that will remove the need for an "init container" for the addon.

currently it will check if a CR already exists. If not it will create one.

/hold

### CODE REVIEW

**Comment by sagidayan** [comment]
/hold

Want to check something tomorrow before merging

File: main.go
Comment (mresvanis) [inline main.go:205]: I think it would read a bit easier if this if-else would return early:

```go
if err != nil && !isNotFound {
  return err
}
```

File: main.go
Comment (mresvanis) [inline main.go:217]: IMHO it's a bit cleaner to only from `main`  and wrap the current error with the information we need to pass on. For example:

```go
return fmt.Errorf("failed to fetch GPUAddon CR %s in %s: %w", common.GlobalConfig.AddonID, common.GlobalConfig.AddonNamespace, err)
```

Comment (sagidayan) [reply main.go:217]: Agree, Changing

File: main.go
Comment (mresvanis) [inline main.go:208]: There is no need to initialize the GPUAddon struct here, we can just add the essential `ObjectMeta` fields and all the rest can be specified in the mutate function arg of the `CreateOrPatch`. For example:

```go
gpuAddon.ObjectMeta = metav1.ObjectMeta{
    Name:           common.GlobalConfig.AddonID,
    Namespace: common.GlobalConfig.AddonNamespace,
}

resp, err := controllerutil.CreateOrPatch(context.TODO(), client, gpuAddon, func() error {
		versionLabel := fmt.Sprintf("%v-version", common.GlobalConfig.AddonLabel)

		gpuAddon.ObjectMeta.Labels = map[string]string{
		    versionLabel: version.Version(),
		}

		gpuAddon.Spec = nvidiav1alpha1.GPUAddonSpec{
		    ConsolePluginEnabled: true,
		}

		return nil
	})


```

File: main_test.go
Comment (mresvanis) [inline main_test.go:7]: Do we specifically need the `olmscheme` here? If there is no specific reason, we can just use the k8s scheme `"k8s.io/client-go/kubernetes/scheme"`.

File: main.go
Comment (sagidayan) [reply main.go:208]: I see your point. And will change, but I do not wish to override the Spec on update. i.e if the console plugin was turned off - it should stay that way.

Thanks for the input.

**Review by mresvanis — APPROVED** [review]

**Pull request merged**
